framework/src/processing/lib/spark-runtime/emr-containers/emr-eks-cluster-props.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+
+import { RemovalPolicy } from 'aws-cdk-lib';
+import { IVpc } from 'aws-cdk-lib/aws-ec2';
+import {
+  Cluster,
+  KubernetesVersion,
+} from 'aws-cdk-lib/aws-eks';
+import { ILayerVersion } from 'aws-cdk-lib/aws-lambda';
+
+/**
+ * The properties for the EmrEksCluster Construct class.
+ */
+export interface SparkEmrContainersRuntimeProps {
+  /**
+   * Name of the Amazon EKS cluster to be created
+   * @default -  The [default cluster name]{@link DEFAULT_CLUSTER_NAME}
+   */
+  readonly eksClusterName?: string;
+  /**
+   * Amazon IAM Role to be added to Amazon EKS master roles that will give access to kubernetes cluster from AWS console UI.
+   * An admin role must be passed if `eksCluster` property is not set.
+   * You will use this role to grant other access to and manage EKS cluster
+   */
+  readonly eksAdminRoleArn: string;
+  /**
+   * The EKS cluster to setup EMR on. The cluster needs to be created in the same CDK Stack.
+   * If the EKS cluster is provided, the cluster AddOns and all the controllers (Ingress controller, Cluster Autoscaler or Karpenter...) need to be configured.
+   * When providing an EKS cluster, the methods for adding nodegroups can still be used. They implement the best practices for running Spark on EKS.
+   * @default - An EKS Cluster is created
+   */
+  readonly eksCluster?: Cluster;
+  /**
+   * Kubernetes version for Amazon EKS cluster that will be created
+   * The default is changed as new version version of k8s on EKS becomes available
+   * @default -  Kubernetes version {@link DEFAULT_EKS_VERSION}
+   */
+  readonly kubernetesVersion?: KubernetesVersion;
+  /**
+   * If set to true, the Construct will create default EKS nodegroups or node provisioners (based on the autoscaler mechanism used).
+   * There are three types of nodes:
+   *  * Nodes for critical jobs which use on-demand instances, high speed disks and workload isolation
+   *  * Nodes for shared worklaods which uses spot instances and no isolation to optimize costs
+   *  * Nodes for notebooks which leverage a cost optimized configuration for running EMR managed endpoints and spark drivers/executors.
+   * @default -  true
+   */
+  readonly defaultNodes?: boolean;
+  /**
+   * The version of karpenter to pass to Helm
+   * @default - The [default Karpenter version]{@link DEFAULT_KARPENTER_VERSION}
+   */
+  readonly karpenterVersion?: string;
+  /**
+   * Starting k8s 1.22, CDK no longer bundle the kubectl layer with the code due to breaking npm package size.
+   * A layer needs to be passed to the Construct.
+   *
+   * The cdk [documentation] (https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.KubernetesVersion.html#static-v1_22)
```

Comment:
```suggestion
   * The cdk [documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.KubernetesVersion.html#static-v1_22)
```
Also not sure this kind of links works, did you check the API.md ? It probably works but better to check.